docs/content/docs/tutorials/emr-on-eks-example.md

```diff
@@ -0,0 +1,494 @@
+```
+OIDC_PROVIDER=$(aws eks describe-cluster --name $EKS_CLUSTER_NAME --region $AWS_REGION --query "cluster.identity.oidc.issuer" --output text | sed -e "s/^https:\/\///")
+ACK_K8S_SERVICE_ACCOUNT_NAME=ack-$SERVICE-controller
+
+echo "creating ${ACK_CONTROLLER_IAM_ROLE}"
+read -r -d '' TRUST_RELATIONSHIP <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${ACK_K8S_NAMESPACE}:${ACK_K8S_SERVICE_ACCOUNT_NAME}"
+        }
+      }
+    }
+  ]
+}
+EOF
+echo "${TRUST_RELATIONSHIP}" > trust.json
+```
+```
+ACK_CONTROLLER_IAM_ROLE="ack-${SERVICE}-controller"
+ACK_CONTROLLER_IAM_ROLE_DESCRIPTION="IRSA role for ACK ${SERVICE} controller deployment on EKS cluster using Helm charts"
+aws iam create-role --role-name "${ACK_CONTROLLER_IAM_ROLE}" \
+    --assume-role-policy-document file://trust.json \
+    --description "${ACK_CONTROLLER_IAM_ROLE_DESCRIPTION}"
+```
+Once the IAM role is created, we can secure it by annotating to EMR on EKS controller service account
+```
+export ACK_K8S_NAMESPACE=ack-system
+export ACK_K8S_SERVICE_ACCOUNT_NAME=ack-$SERVICE-controller
+export ACK_CONTROLLER_IAM_ROLE_ARN=$(aws iam get-role --role-name=$ACK_CONTROLLER_IAM_ROLE --query Role.Arn --output text)
+export IRSA_ROLE_ARN=eks.amazonaws.com/role-arn=$ACK_CONTROLLER_IAM_ROLE_ARN
+```
+```
+echo "annotating ${SERVICE}-controller serviceaccount"
+kubectl annotate serviceaccount -n $ACK_K8S_NAMESPACE $ACK_K8S_SERVICE_ACCOUNT_NAME $IRSA_ROLE_ARN
+
+# Note the deployment name for ACK service controller from following command
+echo "restarting ${SERVICE}-controller pods"
+kubectl get deployments -n $ACK_K8S_NAMESPACE
+kubectl -n $ACK_K8S_NAMESPACE rollout restart deployment ack-emrcontainers-controller-emrcontainers-chart
+
+# validate annotation
+kubectl get pods -n $ACK_K8S_NAMESPACE
+echo "sleeping for 5sec for pod to get ready"
+sleep 5
+CONTROLLER_POD_NAME=$(kubectl get pods -n $ACK_K8S_NAMESPACE --selector=app.kubernetes.io/name=emrcontainers-chart -o jsonpath='{.items..metadata.name}')
+kubectl describe pod -n $ACK_K8S_NAMESPACE $CONTROLLER_POD_NAME | grep "^\s*AWS_"
+```
+**Expected outcome**
+```
+AWS_REGION:                      us-west-2
+AWS_ENDPOINT_URL:                
+AWS_ROLE_ARN:                    arn:aws:iam::012345678910:role/ack-emrcontainers-controller
+AWS_WEB_IDENTITY_TOKEN_FILE:     /var/run/secrets/eks.amazonaws.com/serviceaccount/token (http://eks.amazonaws.com/serviceaccount/token)
+```
+Download the recommended managed policies and apply them to the newly created IRSA role
+```
+cat <<EOF > controller_policy.json
```

Comment:
There are multiple voluminous IAM policies in this tutorial. Maybe we can put them somewhere in emcrcontainer-repository and use curl combined with aws CLI?

Reply:
Or maybe somewhere near the tutorial MD file?

Reply:
Ideally that's what the `recommended-policy-arn` (or `recommended-inline-policy`) should be doing - https://github.com/aws-controllers-k8s/emrcontainers-controller/blob/main/config/iam/recommended-policy-arn . Do we need to adjust this file, so that users can simply reference the value inside here, rather than using the one given below?

Reply:
Yep, I have it in my to-do list to add recommended-policy into controller config. Current arn lacks tag level policy but I can add the policy to recommended-inline-policy. I'll create Github issue to track this
Good to see you @RedbackThomson :blush:

Reply:
https://github.com/aws-controllers-k8s/community/issues/1468